langchain/vectorstores/azuresearch.py

```diff
@@ -0,0 +1,484 @@
+                        prioritized_fields=PrioritizedFields(
+                            title_field=SemanticField(field_name=FIELDS_TITLE),
+                            prioritized_keywords_fields=[
+                                SemanticField(field_name=FIELDS_TAG)
+                            ],
+                            prioritized_content_fields=[
+                                SemanticField(field_name=FIELDS_CONTENT)
+                            ],
+                        ),
+                    )
+                ]
+            )
+        )
+        # Create the search index with the semantic settings and vector search
+        index = SearchIndex(
+            name=index_name,
+            fields=fields,
+            vector_search=vector_search,
+            semantic_settings=semantic_settings,
+        )
+        index_client.create_index(index)
+    # Create the search client
+    return SearchClient(
+        endpoint=endpoint, index_name=index_name, credential=AzureKeyCredential(key)
+    )
+
+
+class AzureSearch(VectorStore):
+    def __init__(
+        self,
+        azure_cognitive_search_name: str,
+        azure_cognitive_search_key: str,
+        index_name: str,
+        embedding_function: Callable,
+        semantic_configuration_name: Optional[str] = None,
+        semantic_query_language: str = "en-us",
+        **kwargs: Any,
+    ):
+        """Initialize with necessary components."""
+        # Initialize base class
+        self.embedding_function = embedding_function
+        self.client = get_search_client(
+            azure_cognitive_search_name,
+            azure_cognitive_search_key,
+            index_name,
+            semantic_configuration_name,
+        )
+        self.semantic_configuration_name = semantic_configuration_name
+        self.semantic_query_language = semantic_query_language
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Add texts data to an existing index."""
+        keys = kwargs.get("keys")
+        ids = []
+        # Write data to index
+        data = []
+        for i, text in enumerate(texts):
+            # Use provided key otherwise use default key
+            key = keys[i] if keys else str(uuid.uuid4())
+            metadata = metadatas[i] if metadatas else {}
+            # Add data to index
+            data.append(
+                {
+                    "@search.action": "upload",
+                    FIELDS_ID: key,
+                    FIELDS_TITLE: metadata.get(FIELDS_TITLE, ""),
+                    FIELDS_TAG: metadata.get(FIELDS_TAG, ""),
+                    FIELDS_CONTENT: text,
+                    FIELDS_CONTENT_VECTOR: np.array(
+                        self.embedding_function(text), dtype=np.float32
+                    ).tolist(),
+                    FIELDS_METADATA: json.dumps(metadata),
```

Comment:
Agree, this is in line with other vector stores implementations but not the best.

Reply:
I'm already finding in one of the projects that I really want to filter on just one of the json fields in the metadata.  This doesn't make any sense within the capabilities of ACS filtering and searching, I might suggest that it likely doesn't in some of the other implementations as well and this may be an opportunity to change this langchain pattern, but that is probably better done in a different PR, otherwise this one will never get merged.